src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrTests.cs

```diff
@@ -0,0 +1,376 @@
+        await CheckAllIsCommited(vmrPath);
+
+        // Add a file in the submodule
+
+        File.WriteAllText(Path.Combine(externalRepoPath, "additional-file.txt"), "New external repo file");
+        await CommitAll(externalRepoPath, "Adding new file in the submodule");
+
+        await processManager.ExecuteGit(submodulePath, new string[] { "pull", "origin", "main" }, CancellationToken.None);
+        await CommitAll(testRepoPath, "Checkout submodule");
+        commit = await GetRepoLastCommit(testRepoPath);
+
+        res = await processManager.Execute(darcExecutable, new string[] { "vmr", "update", "--vmr", vmrPath, "--tmp", tmpPath, $"test-repo:{commit}" });
+        res.ExitCode.Should().Be(0);
+
+        expectedFiles = new List<string>
+        {
+            Path.Combine(vmrPath, "git-info", "AllRepoVersions.props"),
+            Path.Combine(vmrPath, "git-info", "test-repo.props"),
+            Path.Combine(vmrPath, "src", "source-manifest.json"),
+            Path.Combine(vmrPath, "src", "source-mappings.json"),
+            Path.Combine(vmrPath, "src", "test-repo", ".gitmodules"),
+            Path.Combine(vmrPath, "src", "test-repo", "externals", "external-repo", "additional-file.txt"),
+            Path.Combine(vmrPath, "src", "test-repo", "externals", "external-repo", "external-repo-file.txt"),
+            Path.Combine(vmrPath, "src", "test-repo", "test-repo-file.txt")
+        };
+
+        CheckDirectoryContents(new DirectoryInfo(vmrPath), expectedFiles);
+        await CheckAllIsCommited(vmrPath);
+
+        // Remove submodule
+
+        await RemoveSubmodule(testRepoPath, submodulePath);
+        await CommitAll(testRepoPath, "Remove the submodule");
+        commit = await GetRepoLastCommit(testRepoPath);
+
+        res = await processManager.Execute(darcExecutable, new string[] { "vmr", "update", "--vmr", vmrPath, "--tmp", tmpPath, $"test-repo:{commit}" });
+        res.ExitCode.Should().Be(0);
+
+        expectedFiles = new List<string>
+        {
+            Path.Combine(vmrPath, "git-info", "AllRepoVersions.props"),
+            Path.Combine(vmrPath, "git-info", "test-repo.props"),
+            Path.Combine(vmrPath, "src", "source-manifest.json"),
+            Path.Combine(vmrPath, "src", "source-mappings.json"),
+            Path.Combine(vmrPath, "src", "test-repo", ".gitmodules"),
+            Path.Combine(vmrPath, "src", "test-repo", "test-repo-file.txt")
+        };
+
+        CheckDirectoryContents(new DirectoryInfo(vmrPath), expectedFiles);
+        await CheckAllIsCommited(vmrPath);
+    }
+
+    private void CheckDirectoryContents(DirectoryInfo directory, IList<string> expectedFiles)
+    {
+        var filesInDir = GetAllFilesInDirectory(directory).OrderBy(x => x).ToList();
+        filesInDir.Count.Should().Be(expectedFiles.Count);
+        for(int i = 0; i < expectedFiles.Count; i++)
+        {
+            filesInDir[i].Should().Be(expectedFiles[i]);
+        }
```

Comment:
There is a method such as `Should().BeEquivalent` or `BeSame` that compares collections. Check FluentAssertion docs or google